sdk/keyvault/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,122 @@
+* [Troubleshooting Azure Key Vault Secrets SDK Issues](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/keyvault/keyvault-secrets/TROUBLESHOOTING.md)
+
+## Table of Contents
+
+* [Troubleshooting Authentication Issues](#troubleshooting-authentication-issues)
+  * [HTTP 401 Errors](#http-401-errors)
+    * [Frequent HTTP 401 Errors in Logs](#frequent-http-401-errors-in-logs)
+    * [AKV10032: Invalid issuer](#akv10032-invalid-issuer)
+    * [Other Authentication Issues](#other-authentication-issues)
+  * [HTTP 403 Errors](#http-403-errors)
+    * [Operation Not Permitted](#operation-not-permitted)
+* [Other Service Errors](#other-service-errors)
+  * [HTTP 429: Too Many Request](#http-429-too-many-requests)
+
+## Troubleshooting Authentication Issues
+
+### HTTP 401 Errors
+
+HTTP 401 errors may indicate problems authenticating.
+
+#### Frequent HTTP 401 Errors in Logs
+
+Most often, this is expected. Azure Key Vault issues a challenge for initial requests that force authentication. You may see these errors most often during application startup, but may also see these periodically during the application's lifetime when authentication tokens are near expiration.
+
+If you are not seeing subsequent exceptions from the Key Vault SDKs, authentication challenges are likely the cause.
+
+#### AKV10032: Invalid issuer
+
+You may see an error similar to:
+
+```text
+
+RestError: Service request failed.
+Status: 401 (Unauthorized)
+Content:
+{"error":{"code":"Unauthorized","message":"AKV10032: Invalid issuer. Expected one of https://sts.windows.net/{tenant 1}/, found https://sts.windows.net/{tenant 2}/."}}
+
+```
+
+This is most often caused by being logged into a different tenant than the Key Vault authenticates.
+
+See our [DefaultAzureCredential] documentation to see the order credentials are read. You may be logged into a different
+
+tenant for one credential that gets read before another credential. For example, you might be logged into Visual Studio
+
+under the wrong tenant even though you're logged into the Azure CLI under the right tenant.
+
+Automatic tenant discovery support has been added when referencing package `@azure/identity` version
+2.0.0 or newer, and any of the following Key Vault SDK package versions or newer:
+
+Package | Minimum Version
+--- | ---
+
+`@azure/keyvault-admin` | 4.2.0
+`@azure/keyvault-certificates` | 4.4.0
+`@azure/keyvault-keys` | 4.4.0
+`@azure/keyvault-secrets` | 4.4.0
+
+Upgrading to the package versions should resolve any "Invalid Issuer" errors as long as the application or user is a member of the resource's tenant.
+
+#### Other Authentication Issues
+
+If you are using the `@azure/identity` package - which contains [DefaultAzureCredential] - to authenticate requests to Azure Key Vault, please see our [troubleshooting guide](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/Azure.Identity/Troubleshooting.md).
```

Comment:
That URL isn't yet live. @sadasant any ETA?

Validation may fail since the link will 404. Depends on which validation tools are running against your PRs. You may need a placeholder.

Reply:
The url is https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/Troubleshooting.md 🙂 I didn't notice it either! Thank you for pointing it out

Reply:
Oh it's live - this and other missing links (as well as the extraneous blank lines) are just copy-pasta failures :) I'll fix them shortly.

Reply:
it's just that it should be https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/Troubleshooting.md

Reply:
@sadasant you can double check with @joshfree but I think we want to go for `TROUBLESHOOTING.md` consistency across repos so it's easy to reference/discover (since GitHub is case-sensitive). Is it referenced many places, or could we update it in this PR for consistency?

Reply:
I mean, [everybody's doing it](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/TROUBLESHOOTING.md).

/cc @schaabs

Reply:
"I'm happy to bundle these changes" vs. "I just won't be doing the actual work" 😕

But, yeah, I'd say may as well and fix them both up. The sooner it's updated, the less likely someone will 404 that may have already saved the link.

Reply:
> "I'm happy to bundle these changes" vs. "I just won't be doing the actual work" 😕
> 
> But, yeah, I'd say may as well and fix them both up. The sooner it's updated, the less likely someone will 404 that may have already saved the link.

ha, I was trying to make a joke. They won't all be winners, but we have to keep trying 😄 

I switched to using the aka.ms link here instead, one less place to change when we rename this. @sadasant up to you - I am happy to rename the identity file in this PR, we will just need to update the aka.ms links to point to the new URL but just making sure there's not some reason _not to_ rename it

Reply:
Well we have a bunch of aka.ms that point to the older file so I'd rather not rename the real file as part of this PR. We'll be working on the rename this week.